caikit_nlp/toolkit/text_generation/tgis_utils.py

```diff
@@ -41,20 +48,53 @@
         Whether or not the source string should be contained in the generated output,
         e.g., as a prefix.
     input_tokens: bool
-        Whether or not to include list of input tokens.  
+        Whether or not to include list of input tokens.
     generated_tokens: bool
-        Whether or not to include list of individual generated tokens.  
+        Whether or not to include list of individual generated tokens.
     token_logprobs: bool
-        Whether or not to include logprob for each returned token.  
+        Whether or not to include logprob for each returned token.
         Applicable only if generated_tokens == true and/or input_tokens == true
     token_ranks: bool
-        Whether or not to include rank of each returned token.     
-        Applicable only if generated_tokens == true and/or input_tokens == true                             
+        Whether or not to include rank of each returned token.
+        Applicable only if generated_tokens == true and/or input_tokens == true
 """.format(
     GENERATE_FUNCTION_ARGS
 )
 
 
+# Mapping from grpc status codes to caikit status codes. There is not a 1:1
+# mapping at the moment, so this conversion is lossy!
+GRPC_TO_CAIKIT_CORE_STATUS = {
+    grpc.StatusCode.CANCELLED: CaikitCoreStatusCode.CONNECTION_ERROR,
+    grpc.StatusCode.UNKNOWN: CaikitCoreStatusCode.UNKNOWN,
+    grpc.StatusCode.INVALID_ARGUMENT: CaikitCoreStatusCode.INVALID_ARGUMENT,
+    grpc.StatusCode.DEADLINE_EXCEEDED: CaikitCoreStatusCode.CONNECTION_ERROR,
+    grpc.StatusCode.NOT_FOUND: CaikitCoreStatusCode.NOT_FOUND,
+    grpc.StatusCode.ALREADY_EXISTS: CaikitCoreStatusCode.INVALID_ARGUMENT,
+    grpc.StatusCode.PERMISSION_DENIED: CaikitCoreStatusCode.FORBIDDEN,
+    grpc.StatusCode.RESOURCE_EXHAUSTED: CaikitCoreStatusCode.UNKNOWN,
+    grpc.StatusCode.FAILED_PRECONDITION: CaikitCoreStatusCode.UNKNOWN,
+    grpc.StatusCode.ABORTED: CaikitCoreStatusCode.CONNECTION_ERROR,
+    grpc.StatusCode.OUT_OF_RANGE: CaikitCoreStatusCode.UNKNOWN,
```

Comment:
Conversions are lossy, but kind of make sense from changing the view of runtime to library.... For some of these, like the `OUT_OF_RANGE`, might make sense to expose at library level as well ?  Or may be we can map it to `FAILED_PRE_CONDITION` since there is a overlap as per doc [here](https://grpc.github.io/grpc/core/md_doc_statuscodes.html)

Also the `FAILED_PRECONDITION`  might come in handy, which is getting mapped to `UNKNOWN` above 🤔

Reply:
Yeah, I was trying to make this work without changing `caikit` itself. I think there is almost certainly an important API-breaking change conversation to be had about how the internal error codes map to well-known error code enums (`grpc`/`http`) and how to avoid information loss.